contracts/TokenConverter/AbstractTokenConverter.sol

```diff
@@ -768,15 +899,30 @@ abstract contract AbstractTokenConverter is AccessControlledV8, IAbstractTokenCo
         uint256 tokenInUnderlyingPrice = priceOracle.getPrice(tokenAddressIn);
         uint256 tokenOutUnderlyingPrice = priceOracle.getPrice(tokenAddressOut);
 
+        uint256 incentive = configuration.incentive;
+        if (IConverterNetwork(converterNetwork).isTokenConverter(msg.sender)) {
+            incentive = 0;
+        }
+
         /// amount of tokenAddressOut after including incentive
-        uint256 conversionWithIncentive = MANTISSA_ONE + configuration.incentive;
+        uint256 conversionWithIncentive = MANTISSA_ONE + incentive;
         /// conversion rate after considering incentive(conversionWithIncentive)
         tokenInToOutConversion = (tokenInUnderlyingPrice * conversionWithIncentive) / tokenOutUnderlyingPrice;
 
         amountInMantissa = ((amountOutMantissa * EXP_SCALE) + tokenInToOutConversion - 1) / tokenInToOutConversion; //round-up
         amountConvertedMantissa = amountOutMantissa;
     }
 
+    /// @notice Check if msg.sender is allowed to convert as per onlyForPrivateConversions flag
+    /// @param tokenAddressIn Address of the token to convert
+    /// @param tokenAddressOut Address of the token to get after conversion
+    function _checkPrivateConversion(address tokenAddressIn, address tokenAddressOut) internal view {
```

Comment:
`_checkPrivateConversion` is specially designed for users if `onlyForPrivateConversions` is true then it will revert only.
Because if any converter is doing `privateConversion` we cannot **revert** that transaction in any manner as the transaction would contain all the transfers from PSR to all converters up to destinationAddresses for the same. we can only skip the `privateConversion` (if we want to) for any converter:

Since converters can only do the `privateConversions` and they can only get the addresses for the other converters through `ConverterNetwork`, instead of reverting here in any converter through this check which means as per the proposal i.e.
`converterNetwork == address(0) || !IConverterNetwork(converterNetwork).isTokenConverter(msg.sender)`
which means `converterNetwork` is zero address means this converter is not available for `privateConversions` OR `msg.sender` is a user(not converter).

I think we should filter out the converters addresses in the `converterNetwork`